assets/app/scripts/controllers/logs/buildLog.js

```diff
@@ -0,0 +1,80 @@
+'use strict';
+
+angular.module('openshiftConsole')
+  .controller('BuildLog', [
+    '$anchorScroll',
+    '$location',
+    '$q',
+    '$routeParams',
+    '$scope',
+    '$timeout',
+    '$window',
+    'DataService',
+    'project',
+    'logLinks',
+    function($anchorScroll, $location, $q, $routeParams, $scope, $timeout, $window, DataService, project, logLinks) {
+
+      DataService.list("projects", $scope, function(projects) {
+        $scope.projects = projects.by("metadata.name");
+        Logger.log("projects", $scope.projects);
+      });
+
+      project
+        .get($routeParams.project)
+        .then(_.spread(function(project, context) {
+          return $q.all([
+                    DataService
+                      .get('builds', $routeParams.build, context),
+                    DataService
+                      .get('builds/log', $routeParams.build, context)
+                  ])
+                  .then(_.spread(function(build, log) {
```

Comment:
This means we have to wait for the whole log to load before we even get just the build details showing up? Not sure I like that.